src/core/bittorrent/session.cpp

```diff
@@ -2111,6 +2122,9 @@ void Session::handleAddTorrentAlert(libtorrent::add_torrent_alert *p)
             }
         }
 
+        if (pref->isAddTrackersEnabled() && !torrent->isPrivate())
+            torrent->addTrackers(m_additionalTrackers);
```

Comment:
In General I agree with @sledgehammer999. But...
You have not considered the case of preload magnet links. So place ```p.trackers += m_additionalTrackers;``` in Session::loadMetadata().
And put this line below in addTorrent_impl() to make it work on all added torrents (not only for magnet links).